plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
This is generally handled by the sql.DB no? it shouldn't continue to use a bad connection. there was a bug regarding this ages ago iirc. https://github.com/golang/go/issues/3777

either way that problem is solved by setting the maxlifetime value. (see you're above question on why it should be included)